src/control/common/storage/commands.go

```diff
@@ -53,14 +66,41 @@ type StoragePrepareCmd struct {
 	Force    bool `short:"f" long:"force" description:"Perform format without prompting for confirmation"`
 }
 
-func (cmd *StoragePrepareCmd) Init(log logging.Logger) error {
+// Validate checks both only options are not set and returns flags to direct
+// which subsystem types to prepare.
+func (cmd *StoragePrepareCmd) Validate() (bool, bool, error) {
+	prepNvme := cmd.NvmeOnly || !cmd.ScmOnly
+	prepScm := cmd.ScmOnly || !cmd.NvmeOnly
+
 	if cmd.NvmeOnly && cmd.ScmOnly {
-		return errors.New("nvme-only and scm-only options should not be set together")
+		return false, false, errors.New(
+			"nvme-only and scm-only options should not be set together")
 	}
 
-	log.Info(MsgStoragePrepareWarn)
+	return prepNvme, prepScm, nil
+}
+
+func (cmd *StoragePrepareCmd) CheckWarn(state ScmState) error {
+	switch state {
+	case ScmStateNoRegions:
+		if cmd.Reset {
+			return nil
+		}
+	case ScmStateFreeCapacity, ScmStateNoCapacity:
+		if !cmd.Reset {
+			return nil
+		}
+	case ScmStateUnknown:
```

Comment:
There should be a default case here, just in case a new state gets added and someone forgets to update this switch. And the error in the default case would be better as `errors.Errorf("unhandled scm state %q", state)` since it implements fmt.Stringer.

Reply:
addressed in https://github.com/daos-stack/daos/pull/1075